src/Microsoft.DocAsCode.Dfm/CodeRenderer/DfmCodeRenderer.cs

```diff
@@ -39,27 +40,38 @@ public virtual StringBuffer Render(IMarkdownRenderer renderer, DfmFencesToken to
 
             try
             {
-                // Always report original dependency
-                context.ReportDependency(token.Path);
-
                 var pathQueryOption =
                     !string.IsNullOrEmpty(token.QueryStringAndFragment) ?
                     _dfmCodeExtractor.ParsePathQueryString(token.QueryStringAndFragment) :
                     null;
                 var filePath = FindFile(token, context);
+
+                // Always report original dependency
+                context.ReportDependency(token.Path);
+
                 var code = ExtractCode(token, filePath, pathQueryOption);
                 return RenderFencesCode(token, renderer.Options, code.ErrorMessage, code.CodeLines, pathQueryOption);
             }
             catch (DirectoryNotFoundException)
             {
+                if (isValidPath(token.Path))
+                {
+                    context.ReportDependency(token.Path);
+                }
                 return RenderReferenceNotFoundErrorMessage(renderer, token);
             }
             catch (FileNotFoundException)
             {
+                if (isValidPath(token.Path))
+                {
+                    context.ReportDependency(token.Path);
+                }
                 return RenderReferenceNotFoundErrorMessage(renderer, token);
             }
         }
 
+        public bool isValidPath(string path) => !_invalidChars.Any(c => path.Contains(c));
```

Comment:
Can we reuse `PathUtility.IsValidFilePath`?

Reply:
As I tested, it's working well